src/envoy/http/jwt_auth/jwt_authenticator.cc

```diff
@@ -196,10 +196,15 @@ void JwtAuthenticator::VerifyKey(const PubkeyCacheItem& issuer_item) {
   }
 
   headers_->addReferenceKey(kJwtPayloadKey, jwt_->PayloadStrBase64Url());
-  const LowerCaseString key(issuer_item.jwt_config().forward_payload_header());
-  if (!key.get().empty() && !jwt_->PayloadStrBase64Url().empty() &&
-      !(key == kJwtPayloadKey)) {
-    headers_->addCopy(key, jwt_->PayloadStrBase64Url());
+
+  if (issuer_item.jwt_config().forward_payload_header().empty() == false) {
```

Comment:
I felt it is easier to read if we do  

if (!empty)  

 than 

if   empty == false

Reply:
Done.